src/Utility/ToolsUtility.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+using OpenAI.Embeddings;
+
+namespace OpenAI;
+
+internal static class ToolsUtility
+{
+    internal static readonly BinaryData EmptyParameters = BinaryData.FromString("""{ "type" : "object", "properties" : {} }""");
+    internal const string McpToolSeparator = "_-_";
+
+    internal static string GetMethodDescription(MethodInfo method)
+    {
+        var description = method.Name;
+        var attr = method.GetCustomAttribute<DescriptionAttribute>();
+        if (attr != null)
+            description = attr.Description;
+        return description;
+    }
+
+    internal static string GetParameterDescription(ParameterInfo param)
+    {
+        string description = param.Name!;
+        var attr = param.GetCustomAttribute<DescriptionAttribute>();
+        if (attr != null)
+            description = attr.Description;
+        return description;
+    }
+
+    internal static ReadOnlySpan<byte> ClrToJsonTypeUtf8(Type clrType) =>
+        clrType switch
+        {
+            Type t when t == typeof(double) => "number"u8,

```

Comment:
we should add more types here

Reply:
we should also write a good doc remarks describing the type system we support.

Reply:
added more types - will doc in a follow up PR.